extensions/core/commandlog/applib/src/main/java/org/apache/isis/extensions/commandlog/applib/subscriber/CommandSubscriberForCommandLog.java

```diff
@@ -47,14 +48,14 @@
 public class CommandSubscriberForCommandLog implements CommandSubscriber {
 
     final CommandLogEntryRepository<? extends CommandLogEntry> commandLogEntryRepository;
+    final IsisConfiguration isisConfiguration;
 
     @Override
     public void onCompleted(final Command command) {
 
-        // TODO: JPA does not yet support lifecycle listeners, so always would be false.
-//        if(!command.isSystemStateChanged()) {
-//            return;
-//        }
+        if (isisConfiguration.getExtensions().getCommandLog().getPublishPolicy().isOnlyIfSystemChanged() && !command.isSystemStateChanged()) {
```

Comment:
TODO this logic seems wrong, at least I'm confused reading this, maybe amend with comments